src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -322,7 +339,8 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
             {
                 if (accessToken.ClientId.Equals(ClientId, StringComparison.OrdinalIgnoreCase) &&
                     environmentAliases.Contains(accessToken.Environment) &&
-                    accessToken.TenantId.Equals(tenantId, StringComparison.OrdinalIgnoreCase) &&
+                    (accessToken.IsAdfs ||
+                    accessToken.TenantId.Equals(tenantId, StringComparison.OrdinalIgnoreCase)) &&
```

Comment:
what is the TenantId in ADFS scenarios?

Reply:
null. Adfs doesn't have a concept of tenants